server/src/main/java/org/apache/druid/client/selector/ConnectionCountServerSelectorStrategy.java

```diff
@@ -27,16 +27,20 @@
 import java.util.Comparator;
 import java.util.List;
 import java.util.Set;
+import java.util.stream.Collectors;
 
 public class ConnectionCountServerSelectorStrategy implements ServerSelectorStrategy
 {
-  private static final Comparator<QueryableDruidServer> COMPARATOR =
-      Comparator.comparingInt(s -> s.getClient().getNumOpenConnections());
+  private static final Comparator<RemoteDruidServer> COMPARATOR =
+      Comparator.comparingInt(s -> s.getQueryRunner().getNumOpenConnections());
 
   @Override
   public QueryableDruidServer pick(Set<QueryableDruidServer> servers, DataSegment segment)
   {
-    return Collections.min(servers, COMPARATOR);
+    return Collections.min(
+        servers.stream().map(server -> (RemoteDruidServer) server).collect(Collectors.toSet()),
```

Comment:
All I did in this PR is changing the type of `COMPARATOR`. The suggestion sounds good, but I don't think this should be part of this PR.

Reply:
I think it's better if the PR scope is as narrow as possible because it helps reviewers a lot to focus on the main purpose of the PR. Your suggestion sounds good so maybe we can do in another PR, but I don't think it's related to the purpose of this PR.